File: code/02-word-game.py
# 02-word-game.py
# Slide 01: Talking with LLMs via code (Anatomy of a conversation)
# Goal: play a word guessing game where the LLM is the guesser. The first user
# message includes a modifier (e.g. "In British English,") that steers later
# turns; this shows how the conversation carries state in the message history.

# %% Import packages and load environment variable for API access
import chatlas
import dotenv
dotenv.load_dotenv()

# %% Set up a chat with a system prompt
chat = chatlas.ChatAnthropic(
    system_prompt="We are playing a word guessing game. "
    "At each turn, you guess the word and tell us what it is."
)
___("In British English, guess the word for a person who lives next door.")
___("What helps a car move smoothly down the road?")

# %% Compare with...
chat2 = chatlas.ChatAnthropic()
chat2.___

File: code/02-word-game.R
# 02-word-game.R
# Slide 01: Talking with LLMs via code (Anatomy of a conversation)
# Goal: play a word guessing game where the LLM is the guesser. The first user
# message includes a modifier (e.g. "In British English,") that steers later
# turns; this shows how the conversation carries state in the message history.

# %% Import package
library(ellmer)

# %% Set up a chat with a system prompt
chat <- chat_anthropic(
  system_prompt = paste(
    "We are playing a word guessing game.",
    "At each turn, you guess the word and tell us what it is."
  )
)

# Ask the first question:
____("In British English, guess the word for the person who lives next door.")

# Ask the second question:
____("What helps a car move smoothly down the road?")

# %% Compare with...
chat2 <- chat_anthropic()
chat2$____
